ci: Add skipIfOnGHA util

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59748 ci: Add skipIfOnGHA util**

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29008217](https://our.internmc.facebook.com/intern/diff/D29008217)

seemethere committed:
ci: Add skipIfOnGHA and skip foreach tests

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add skipIfOnGHA and skip foreach tests"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29008217).

seemethere commented:
Closing, forward fixing with https://github.com/pytorch/kineto/pull/281

seemethere committed:
Update on "ci: Add skipIfOnGHA and skip foreach tests"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29008217](https://our.internmc.facebook.com/intern/diff/D29008217)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29008217).

walterddr commented on torch/testing/_internal/common_utils.py:
```diff
@@ -391,6 +391,9 @@ def _check_module_exists(name):
 TEST_SKIP_NOARCH = os.getenv('PYTORCH_TEST_SKIP_NOARCH', '0') == '1'
 
+# Disables tests for when on Github Actions
+ON_GHA = os.getenv("GITHUB_ACTIONS")
```
```suggestion
ON_GHA = os.getenv('GITHUB_ACTIONS', '0') == '1'
```

seemethere replied:
changed

seemethere committed:
Update on "ci: Add skipIfOnGHA util"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29008217](https://our.internmc.facebook.com/intern/diff/D29008217)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29008217).
